### PR TITLE
Upgrade to powsybl dependencies 2024.2.0

### PR DESCRIPTION
**Pull request by FranckLecuyer**
Update to gridsuite/dependencies#70

### CODE REVIEW

File: src/main/java/org/gridsuite/study/server/service/ReportService.java [hunk 1]
```diff
@@ -90,7 +91,7 @@ private ReportNode reportServerCall(UUID id, String serverUri, UriComponentsBuil
         var path = uriBuilder.buildAndExpand(id).toUriString();
         HttpHeaders headers = new HttpHeaders();
         headers.setContentType(MediaType.APPLICATION_JSON);
-        List<ReportNode> reporters = restTemplate.exchange(serverUri + path, HttpMethod.GET, new HttpEntity<>(headers), new ParameterizedTypeReference<List<ReportNode>>() {
+        List<ReportNodeImpl> reporters = restTemplate.exchange(serverUri + path, HttpMethod.GET, new HttpEntity<>(headers), new ParameterizedTypeReference<List<ReportNodeImpl>>() {
```
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/study/server/service/ReportService.java:94]: strange, I as persuaded that Jackson handled correctly interfaces when deserializing...

Well just need to update the reference then:
```suggestion
        List<ReportNode> reporters = restTemplate.exchange(serverUri + path, HttpMethod.GET, new HttpEntity<>(headers), new ParameterizedTypeReference<List<ReportNodeImpl>>() {
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 2]
```diff
@@ -181,17 +182,17 @@ public void testReport() {
...
+        checkReportNodes(reports, expectedRootReporter.getChildren().stream().map(r -> (ReportNodeImpl) r).toList());
         assertTrue(TestUtils.getRequestsDone(1, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
 
         mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=false&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), rootNode.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
         reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reports, expectedRootReporter.getChildren());
+        checkReportNodes(reports, expectedRootReporter.getChildren().stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:195]:
```suggestion
        checkReportNodes(reports, expectedRootReporter.getChildren().stream().map(ReportNodeImpl::cast).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 3]
```diff
@@ -181,17 +182,17 @@ public void testReport() {
         MvcResult mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=true&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), rootNode.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reports, expectedRootReporter.getChildren());
+        checkReportNodes(reports, expectedRootReporter.getChildren().stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:187]:
```suggestion
        checkReportNodes(reports, expectedRootReporter.getChildren().stream().map(ReportNodeImpl::cast).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 4]
```diff
@@ -212,7 +213,7 @@ public void testReport() {
         reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
         final List<ReportNode> expectedReports = Stream.concat(expectedRootReporter.getChildren().stream(), expectedNodeReporter.getChildren().stream()).toList();
-        checkReportNodes(reports, expectedReports);
+        checkReportNodes(reports, expectedReports.stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:216]:
```suggestion
        checkReportNodes(reports, expectedReports.stream().map(ReportNodeImpl::cast).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 5]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
         MvcResult mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=true&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), child1.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reports, List.of(expectedChild1Reporter.getChildren().get(1)));
+        checkReportNodes(reports, List.of(expectedChild1Reporter.getChildren().get(1)).stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:244]:
```suggestion
        checkReportNodes(reports, List.of(expectedChild1Reporter.getChildren().get(1)).stream().map(ReportNodeImpl::cast).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 6]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
...
         assertTrue(TestUtils.getRequestsDone(1, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
 
         mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=false&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), child1.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reportsNode1 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reportsNode1 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reportsNode1, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild1Reporter.getChildren().stream()).toList());
+        checkReportNodes(reportsNode1, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild1Reporter.getChildren().stream()).toList().stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:252]:
```suggestion
        checkReportNodes(reportsNode1, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild1Reporter.getChildren().stream()).toList().stream().map(ReportNodeImpl::cast).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 7]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
...
         assertTrue(TestUtils.getRequestsDone(2, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
 
         mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=false&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), child2.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reportsNode2 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reportsNode2 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reportsNode2, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild2Reporter.getChildren().stream()).toList());
+        checkReportNodes(reportsNode2, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild2Reporter.getChildren().stream()).toList().stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:260]:
```suggestion
        checkReportNodes(reportsNode2, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild2Reporter.getChildren().stream()).toList().stream().map(ReportNodeImpl::cast).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 8]
```diff
@@ -273,9 +274,9 @@ public void testSubReport() {
                                 rootNode.getStudyId(), rootNode.getId(), ROOT_NODE_REPORT_UUID.toString()))
                         .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
                         .andReturn();
-        ReportNode report = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        ReportNodeImpl report = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(List.of(report), List.of(expectedRootReporter));
+        checkReportNodes(List.of(report), List.of(expectedRootReporter).stream().map(r -> (ReportNodeImpl) r).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:279]:
```suggestion
        checkReportNodes(List.of(report), List.of((ReportNodeImpl) expectedRootReporter));
```

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/service/ReportService.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/study/server/service/ReportService.java:94]: Same thing has been done in powsybl-core, in this PR : https://github.com/powsybl/powsybl-core/pull/3011
when they upgrade to ReportNode V2.1 ...

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:195]: Done

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java (hunk 3 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:187]: Done

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 9]
```diff
@@ -203,7 +204,7 @@ public void testReport() {
             .andReturn();
         reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reports, expectedNodeReporter.getChildren());
+        checkReportNodes(reports, expectedNodeReporter.getChildren().stream().map(r -> (ReportNodeImpl) r).toList());
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:207]: Done

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java (hunk 4 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:216]: Done

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java (hunk 5 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:244]: Done

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java (hunk 6 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:252]: Done

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java (hunk 7 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:260]: Done

File: src/test/java/org/gridsuite/study/server/utils/TestUtils.java [hunk 10]
```diff
@@ -200,7 +201,7 @@ public static ReportNode addChildReportNode(ReportNode parent, ReportNode child)
         return parent;
     }
 
-    public static void checkReportNodes(List<ReportNode> reports, List<ReportNode> expectedReports) {
+    public static void checkReportNodes(List<ReportNodeImpl> reports, List<ReportNodeImpl> expectedReports) {
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/utils/TestUtils.java:204]: See above

**Review by AbdelHedhili — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/service/ReportService.java (hunk 1 above)
Comment (Tristan-WorkGH) [inline src/main/java/org/gridsuite/study/server/service/ReportService.java:94]:
```suggestion
        List<ReportNode> reporters = restTemplate.exchange(serverUri + path, HttpMethod.GET, new HttpEntity<>(headers), new ParameterizedTypeReference<List<ReportNodeImpl>>() {
```

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 11]
```diff
@@ -1003,7 +1004,7 @@ public void testLogsReport() throws Exception {
         MvcResult mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?reportType=NETWORK_MODIFICATION", studyUuid, rootNodeUuid).header(USER_ID_HEADER, "userId"))
                 .andExpect(status().isOk()).andReturn();
         String resultAsString = mvcResult.getResponse().getContentAsString();
-        List<ReportNode> reporterModel = mapper.readValue(resultAsString, new TypeReference<List<ReportNode>>() { });
+        List<ReportNodeImpl> reporterModel = mapper.readValue(resultAsString, new TypeReference<>() { });
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:1007]:
```suggestion
        List<ReportNode> reporterModel = mapper.readValue(resultAsString, new TypeReference<List<ReportNodeImpl>>() { });
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 12]
```diff
@@ -181,17 +182,17 @@ public void testReport() {
         MvcResult mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=true&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), rootNode.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:185]:
```suggestion
        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<List<ReportNodeImpl>>() {
```

File: src/test/java/org/gridsuite/study/server/utils/TestUtils.java (hunk 10 above)
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/utils/TestUtils.java:204]:
```suggestion
    public static void checkReportNodes(List<ReportNode> reports, List<ReportNode> expectedReports) {
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 13]
```diff
@@ -181,17 +182,17 @@ public void testReport() {
...
         });
-        checkReportNodes(reports, expectedRootReporter.getChildren());
+        checkReportNodes(reports, expectedRootReporter.getChildren().stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:187]:
```suggestion
        checkReportNodes(reports, expectedRootReporter.getChildren());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 14]
```diff
@@ -181,17 +182,17 @@ public void testReport() {
...
         assertTrue(TestUtils.getRequestsDone(1, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
 
         mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=false&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), rootNode.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
         reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reports, expectedRootReporter.getChildren());
+        checkReportNodes(reports, expectedRootReporter.getChildren().stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:195]:
```suggestion
        checkReportNodes(reports, expectedRootReporter.getChildren());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 15]
```diff
@@ -203,7 +204,7 @@ public void testReport() {
             .andReturn();
         reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
-        checkReportNodes(reports, expectedNodeReporter.getChildren());
+        checkReportNodes(reports, expectedNodeReporter.getChildren().stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:207]:
```suggestion
        checkReportNodes(reports, expectedNodeReporter.getChildren());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 16]
```diff
@@ -212,7 +213,7 @@ public void testReport() {
         reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
         });
         final List<ReportNode> expectedReports = Stream.concat(expectedRootReporter.getChildren().stream(), expectedNodeReporter.getChildren().stream()).toList();
-        checkReportNodes(reports, expectedReports);
+        checkReportNodes(reports, expectedReports.stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:216]:
```suggestion
        checkReportNodes(reports, expectedReports.stream());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 17]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
         MvcResult mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=true&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), child1.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:242]:
```suggestion
        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<List<ReportNodeImpl>>() {
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 18]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
...
         });
-        checkReportNodes(reports, List.of(expectedChild1Reporter.getChildren().get(1)));
+        checkReportNodes(reports, List.of(expectedChild1Reporter.getChildren().get(1)).stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:244]:
```suggestion
        checkReportNodes(reports, List.of(expectedChild1Reporter.getChildren());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 19]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
...
         assertTrue(TestUtils.getRequestsDone(1, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
 
         mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=false&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), child1.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reportsNode1 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reportsNode1 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:250]:
```suggestion
        List<ReportNode> reportsNode1 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<List<ReportNodeImpl>>() {
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 20]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
...
         });
-        checkReportNodes(reportsNode1, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild1Reporter.getChildren().stream()).toList());
+        checkReportNodes(reportsNode1, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild1Reporter.getChildren().stream()).toList().stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:252]:
```suggestion
        checkReportNodes(reportsNode1, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild1Reporter.getChildren().stream()).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 21]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
...
         assertTrue(TestUtils.getRequestsDone(2, server).stream().anyMatch(r -> r.matches("/v1/reports/.*")));
 
         mvcResult = mockMvc.perform(get("/v1/studies/{studyUuid}/nodes/{nodeUuid}/parent-nodes-report?nodeOnlyReport=false&reportType=NETWORK_MODIFICATION", rootNode.getStudyId(), child2.getId()))
             .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
             .andReturn();
-        List<ReportNode> reportsNode2 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reportsNode2 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:258]:
```suggestion
        List<ReportNode> reportsNode2 = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<List<ReportNodeImpl>>() {
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 22]
```diff
@@ -238,25 +239,25 @@ public void testMultipleReport() {
...
         });
-        checkReportNodes(reportsNode2, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild2Reporter.getChildren().stream()).toList());
+        checkReportNodes(reportsNode2, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild2Reporter.getChildren().stream()).toList().stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:260]:
```suggestion
        checkReportNodes(reportsNode2, Stream.concat(expectedRootReporter.getChildren().stream(), expectedChild2Reporter.getChildren().stream()).toList());
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 23]
```diff
@@ -273,9 +274,9 @@ public void testSubReport() {
                                 rootNode.getStudyId(), rootNode.getId(), ROOT_NODE_REPORT_UUID.toString()))
                         .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
                         .andReturn();
-        ReportNode report = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        ReportNodeImpl report = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:277]:
```suggestion
        ReportNode report = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<ReportNodeImpl>() {
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 24]
```diff
@@ -273,9 +274,9 @@ public void testSubReport() {
...
         });
-        checkReportNodes(List.of(report), List.of(expectedRootReporter));
+        checkReportNodes(List.of(report), List.of(expectedRootReporter).stream().map(ReportNodeImpl.class::cast).toList());
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:279]:
```suggestion
        checkReportNodes(List.of(report), List.of(expectedRootReporter));
```

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java [hunk 25]
```diff
@@ -289,7 +290,7 @@ public void testNodeReport() {
                                 rootNode.getStudyId(), rootNode.getId(), ROOT_NODE_REPORT_UUID.toString()))
                         .andExpectAll(status().isOk(), content().contentType(MediaType.APPLICATION_JSON))
                         .andReturn();
-        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
+        List<ReportNodeImpl> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<>() {
```
Comment (Tristan-WorkGH) [inline src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:293]:
```suggestion
        List<ReportNode> reports = mapper.readValue(mvcResult.getResponse().getContentAsString(), new TypeReference<List<ReportNodeImpl>>() {
```

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]
To not have cascade change, we only specify the change of powsybl (de)serialization to Jackson, but keep manipulating the interface in our code.

File: src/main/java/org/gridsuite/study/server/service/ReportService.java (hunk 1 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/study/server/service/ReportService.java:94]: ???? This doesn't compile ...

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 11 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/StudyTest.java:1007]: Same as above

File: src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java (hunk 12 above)
Comment (FranckLecuyer) [reply src/test/java/org/gridsuite/study/server/service/ReportServiceTest.java:185]: Same as above

**Review by Tristan-WorkGH — APPROVED** [review]

**Pull request merged**